monai/handlers/nvtx_handlers.py

```diff
@@ -0,0 +1,176 @@
+    Events, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Events")
+
+
+__all__ = ["RangeHandler", "RangePushHandler", "RangePopHandler", "MarkHandler"]
+
+
+class RangeHandler:
+    """
+    Attach a NVTX range to a pair of Ignite events.
+    It pushes an NVTX range at the first event and pops it at the second event.
+    Stores zero-based depth of the range that is started.
+
+    Args:
+        event: a string or pair of Ignite events to attach a range to
+            If a single string is provided, it should  describe the base name of a pair of default Ignite events
+            with _STARTED and _COMPLETED postfix (like "EPOCH" for EPOCH_STARTED and EPOCH_COMPLETED).
+            The common accepted events are: BATCH, ITERATION, EPOCH, and ENGINE.
+            For the complete list of Events,
+            check https://pytorch.org/ignite/generated/ignite.engine.events.Events.html.
+
+        msg: ASCII message to associate with range.
+            If not provided, the name of first event will be assigned to the NVTX range.
+    """
+
+    def __init__(self, events: Union[str, Tuple[Events, Events]], msg: Optional[str] = None) -> None:
+        if msg is None:
+            if isinstance(events, str):
+                msg = events
+            else:
+                msg = events[0].strip("_")[0]
+        self.msg = msg
+        if isinstance(events, str):
+            if events.upper() in ["ENGINE", ""]:
+                event_name = ""
+            elif events.upper() == "BATCH":
+                event_name = "GET_BATCH_"
+            else:
+                event_name = events.upper() + "_"
+            self.events = (
+                getattr(Events, event_name + "STARTED"),
+                getattr(Events, event_name + "COMPLETED"),
+            )
+        elif isinstance(events, Sequence):
+            if len(events) != 2:
+                raise ValueError(f"Exactly two Ignite events should be provided [received {len(events)}].")
+            if not isinstance(events[0], Events):
+                raise ValueError("The provided first event is not an Ignite event!")
+            if not isinstance(events[1], Events):
+                raise ValueError("The provided second event is not an Ignite event!")
+            self.events = events
+        else:
+            raise ValueError("The start/end events should either be a string or tuple/list of two Ignite events.")
+        self.depth = None
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Attach an NVTX Range to specific Ignite events
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(self.events[0], self.range_push)
+        engine.add_event_handler(self.events[1], self.range_pop)
+
+    def range_push(self):
+        self.depth = _nvtx.rangePushA(self.msg)
```

Comment:
May I know what this `depth` member variable will be used for?

Thanks.

Reply:
currently, we are not using this since we already capture this information from Nsight Systems. However, if someone might want to know the depth of the range that is being created so I didn't want to discard this information when the range is being created.